app/Controllers/SettingsController.php

```
@@ -208,6 +210,144 @@ public function updateEmailSettings(Request $request, Response $response, mysqli
+     */
+    public function updateRegistrationSettings(Request $request, Response $response, mysqli $db): Response
+    {
+        $data = (array) $request->getParsedBody();
+        // CSRF validated by CsrfMiddleware
+
+        $repository = new SettingsRepository($db);
+        $repository->ensureTables();
+
+        $requireApproval = isset($data['require_admin_approval']) ? '1' : '0';
+        $repository->set('registration', 'require_admin_approval', $requireApproval);
+        ConfigStore::set('registration.require_admin_approval', (bool) $requireApproval);
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Support\ConfigStore' in method 'updateRegistrationSettings'. (undefined)

(StaticAccess)